R/tools.R

```diff
@@ -70,6 +62,15 @@ tool_matches <- function(tool, labels = NULL) {
   FALSE
 }
 
+# Convert from .btw_tools (or a filtered version of it)
+# to a format compatible with `client$set_tools()`
+as_ellmer_tools <- function(x) {
+  res <- lapply(x, function(.x) {.x$tool()})
+  is_null <- vapply(res, is.null, logical(1))
+  res <- res[!is_null]
+  set_names(res, names(x)[!is_null])
```

Comment:
I think this could also be simplified using the purrr-alike helpers 

```suggestion
  compact(map(x, function(.x) .x$tool()))
```